packages/core/strapi/src/middlewares/security.ts

```diff
@@ -29,6 +29,14 @@ const defaults: Config = {
   },
 };
 
+const mergeConfig = (existingConfig: Config, newConfig: Config) => {
```

Comment:
This kind of merging prevents overwritting completly and should be avoided. 

You can inverse the merge order so that user config will override the default one when provided below

Reply:
@alexandrebodin mergeConfig is only used to update existing user config with extra rules, so it won't overwrite user config, but extend it (set field if missing, and concat array if existing).

Below `helmetConfig` is computed from user `config` and defaults, and won't be affected by this merge strategy.
```
let helmetConfig: Config = defaultsDeep(defaults, config);
```

Please let me know if I'm missing a case, we can have it as test